subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyInsightReportTaskIntegrationTest.groovy

```diff
@@ -585,11 +590,73 @@ org:leaf:1.0 (forced)
 
 org:leaf:1.0
 \\--- org:foo:1.0
-     \\--- conf
+     \\--- forced
 
 org:leaf:2.0 -> 1.0
 \\--- org:bar:1.0
-     \\--- conf
+     \\--- forced
+"""
+
+        when:
+        run "dependencyInsight", "--configuration", "substituted", "--dependency", "leaf"
+
+        then:
+        outputContains """
+org:leaf:1.0 (selected by rule)
+   variant "runtime" [
+      org.gradle.status             = release (not requested)
+      org.gradle.usage              = java-runtime (not requested)
+      org.gradle.component.category = library (not requested)
+   ]
+
+org:leaf:1.0
+\\--- org:foo:1.0
+     \\--- substituted
+
+org:leaf:2.0 -> 1.0
+\\--- org:bar:1.0
+     \\--- substituted
+"""
+    }
+
+    def "shows forced dynamic version"() {
+        given:
+        mavenRepo.module("org", "leaf", "1").publish()
+        mavenRepo.module("org", "leaf", "2").publish()
+        mavenRepo.module("org", "leaf", "3").publish()
+
+        file("build.gradle") << """
+            repositories {
+                maven { url "${mavenRepo.uri}" }
+            }
+            configurations {
+                conf
+            }
+            dependencies {
+                conf('org:leaf:[1,2]') {
+                    because 'testing stuff'
+                    force true
+                }
+            }
+        """
+
+        when:
+        run "dependencyInsight", "--configuration", "conf", "--dependency", "leaf"
+
+        then:
+        outputContains """
+org:leaf:2
+   variant "runtime" [
+      org.gradle.status             = release (not requested)
+      org.gradle.usage              = java-runtime (not requested)
+      org.gradle.component.category = library (not requested)
+   ]
+   Selection reasons:
+      - Was requested : didn't match version 3 because testing stuff
```

Comment:
That reason reads strangely. On first pass I was surprised and then understood why the reason was phrased so.
It feels like updating it to something like what follows would make it somewhat clearer.
> - Was requested : range didn't match version 3 because testing stuff

Reply:
Yes I agree these composed reasons are not great. They have not been changed as part of this PR, but I want to improve them soon.